proto/provenance/marker/v1/tx.proto

```diff
@@ -150,10 +154,39 @@ message MsgTransferResponse {}
 
 // MsgSetDenomMetadataRequest defines the Msg/SetDenomMetadata request type
 message MsgSetDenomMetadataRequest {
+  // metadata is the optional raw metadata structure to be stored.
+  // If not provided, then the description, base_name, and available_prefixes must be provided.
+  // Those will be used to define the appropriate denom metadata structure.
+  //
+  // If provided, it must conform to the following restrictions:
+  // * A DenomUnit must exist with a denom field equal to the base field, and an exponent of 0 (zero).
+  // * A DenomUnit must exist with either a denom field equal to the display field, or an alias equal to the display
+  // field.
+  // * A DenomUnit must exist with an unprefixed denom field and it must not have any aliases.
+  // * The denom field of all DenomUnits must be the long SI prefix + unprefixed denom, e.g. "nanohash" (except the
+  // DenomUnit for the un-prefixed entry).
+  // * The aliases of all DenomUnits must be a single entry of the short SI Prefix + unprefixed denom, e.g. "nhash"
+  // (except the DenomUnit for the un-prefixed entry).
+  // * The exponent of all DenomUnits must be the exponent of the SI prefix on its denom minus the exponent of the SI
+  // prefix on the base DenomUnit. For example, if the base is "nanohash", it's SI prefix exponent is -9 (and the
+  // DenomUnit for it has an exponent of 0). A DenomUnit for "microhash" has an SI prefix exponent of -6. So the
+  // exponent of the "microhash" DenomUnit is -6 - -9 = 3. A DenomUnit for "hash" must have an exponent of 0 - -9 = 9,
+  // and a DenomUnit for "kilohash" must have an exponent of 3 - -9 = 12.
   cosmos.bank.v1beta1.Metadata metadata = 1
-      [(gogoproto.nullable) = false, (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/x/bank/types.Metadata"];
+      [(gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/x/bank/types.Metadata"];
+  // administrator is the address of the administrator making this request.
   string administrator = 2;
+  // description is A description of this denomination.
+  // It is used to populate the Metadata.description field.
+  string description = 3;
+  // base_name is the root of the name of this denomination. SI Prefix strings will be added to this to create the
+  // desired denom units.
+  string base_name = 4;
```

Comment:
This seems like it would be the marker denom itself (as the base_name) and yet the problem with a user does not think to create their denom with the smallest unit of measurement in mind will exist.